code/modules/client/preference_setup/general/04_equipment.dm

```diff
@@ -35,14 +35,14 @@
 	var/undies = get_undies()
 	var/gender_socks = get_gender_socks()
 	if(!get_key_by_value(undies, pref.underwear))
-		pref.underwear = undies[undies[1]]
+		pref.underwear = null
 	if(!get_key_by_value(undershirt_t, pref.undershirt))
 		pref.undershirt = undershirt_t[undershirt_t[1]]
 	if(!get_key_by_value(gender_socks, pref.socks))
-		pref.socks = gender_socks[gender_socks[1]]
-
+		pref.socks = null

```

Comment:
That is costly, though it shouldn't break much. Running sanitize should generally be done _after_ accepting user input. Not before. Logically anyways. Special cases should be handled as they arise, such as the underwear shenanigans with a gender swap, or updating certain records for genders, etc.